pkg/operator/collection.go

```diff
@@ -316,9 +330,24 @@ func (r *collectionReconciler) ensureCollectorConfig(ctx context.Context, spec *
 			Namespace: r.opts.OperatorNamespace,
 			Name:      NameCollector,
 		},
-		Data: map[string]string{
+	}
+
+	switch compression {
+	case monitoringv1.CompressionGzip:
```

Comment:
Let's add comment like 
```suggestion
        // Thanos config-reloader detects gzip compression automatically, so no sync with config-reloaders is needed when switching between those.
	case monitoringv1.CompressionGzip:
```

Reply:
as it might be a bit surprising for future readers.